src/main/java/fi/fmi/avi/model/taf/immutable/TAFImpl.java

```diff
@@ -178,6 +201,217 @@ public Builder withCompleteForecastTimes(final ZonedDateTime reference) {
             return this;
         }
 
+        public Builder withAllTimesComplete(final ZonedDateTime reference) {
+            requireNonNull(reference, "reference");
+            withCompleteIssueTimeNear(reference);
+            return withCompleteForecastTimes(getIssueTime()//
+                    .flatMap(PartialOrCompleteTimeInstant::getCompleteTime)//
+                    .orElse(reference));
+        }
+
+        /**
+         * Sets the TAF-specific message status.
+         *
+         * Note, this method is provided for backward compatibility with previous versions of the API. The <code>status</code> is no longer
+         * explicitly stored. Instead, this method sets other property values with the following logic:
+         * <ul>
+         *     <li>status is {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#CANCELLATION} and reportStatus is not present or is NORMAL:
+         *     <code>reportStatus =</code>
+         *     {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#AMENDMENT}, <code>cancelMessage = true</code></li>
+         *     <li>status is {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#MISSING}: clears the baseForecast if present</li>
+         *     <li>status is {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#NORMAL}: <code>reportStatus =</code>
+         *     {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#NORMAL}, <code>cancelMessage = false</code></li>
+         *     <li>status is {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#AMENDMENT}: <code>reportStatus =</code>
+         *     {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#AMENDMENT}, <code>cancelMessage = false</code></li>
+         *     <li>status is {@link fi.fmi.avi.model.AviationCodeListUser.TAFStatus#CORRECTION}: <code>reportStatus =</code>
+         *     {@link fi.fmi.avi.model.AviationWeatherMessage.ReportStatus#CORRECTION}, <code>cancelMessage = false</code></li>
+         * </ul>
+         *
+         * @param status the status to set
+         * @return builder
+         * @deprecated migrate to using a combination of {@link #setReportStatus(ReportStatus)} and {@link #setCancelMessage(boolean)} instead
+         */
+        @Deprecated
+        public Builder setStatus(final TAFStatus status) {
```

Comment:
As `setStatus()` is now deprecated, invocations to it in [constructor](https://github.com/fmidev/fmi-avi-messageconverter/blob/93b697e3fdf8a376561b23bb3fca544e7809fa55/src/main/java/fi/fmi/avi/model/taf/immutable/TAFImpl.java#L128) and [`from(TAF)`](https://github.com/fmidev/fmi-avi-messageconverter/blob/93b697e3fdf8a376561b23bb3fca544e7809fa55/src/main/java/fi/fmi/avi/model/taf/immutable/TAFImpl.java#L157) should be replaced with invocations to `setCancelMessage()`, `setBaseForecast()` and `setCancelledReportValidPeriod()` (the last two already exist).

Reply:
agreed